src/textual/_region_group.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from operator import attrgetter
+from typing import NamedTuple, Iterable
+
+from src.textual.geometry import Region
+
+
+class InlineRange(NamedTuple):
+    """Represents a region on a single line."""
+
+    line_index: int
+    start: int
+    end: int
+
+
+class RegionGroup:
```

Comment:
On reflection this probably doesn't need to be a class. All we're really interested in is that `inline_ranges` method, which we could extract in to a function that accepts an Iterable of Regions and returns InlineRanges.

We can probably rename it in that case. If it were pixels I'd be tempted to call it `rasterize_regions`. There may be a better term when it comes to characters.

Reply:
Yeah I had the same thought, agreed!

Reply:
Coming up empty for better names for the function - let me know if you have anything.

Reply:
`resolve_regions_to_ranges` ?